torch/distributed/tensor/_dtensor_spec.py

```diff
@@ -32,6 +32,10 @@ class DTensorSpec:
     def __post_init__(self) -> None:
         if not isinstance(self.placements, tuple):
             self.placements = tuple(self.placements)
+        if not len(self.placements) == self.mesh.ndim:
+            raise ValueError(
+                f"DTensorSpec requires one placement per mesh dim (ndim={self.mesh.ndim}), got {self.placements=}"
```

Comment:
`mesh.ndim={self.mesh.ndim)` is better because `ndim` is ambiguous. It can be understood as `DTensorSpec.ndim` which is the tensor_meta ndim.